app/views/documents/show/_tags.html.erb

```diff
@@ -23,7 +18,7 @@
     ),
     items: tags
   } %>
-<% rescue GdsApi::BaseError => e %>
+<% rescue ActionView::Template::Error => e %>
```

Comment:
I'm pretty sure we shouldn't do this, as it'll confuse a future developer who makes a typo in the view. This is definitely the reason we don't catch the API error at the controller level - we would also swallow any other problem with rendering the view.

Another way we could approach this would be to do the API calls in the individual views. We can still have a `multi_tag` partial for the actual rendering, so the result is reasonably DRY. Can we try that out and see how it looks?

To be properly compatible across views, the partials really ought to catch their own exceptions. However, we may as well save ourselves the effort, as we're intending to apply our local data caching approach to the linkables for tags (exception becomes redundant).

Reply:
The API calls are already happening in the individual views. The `GdsApi::BaseError` isn't caught in those views (as we're catching them at this higher level `_tags.html.erb` partial) so Rails throws a `ActionView::Template::Error` error instead.

I could move the `rescue` into each individual view, which I think is what you are suggesting, but there will be a change of behaviour. Instead of using the summary list component to render a "Error connecting to the API" message in its callback block, we'd be rendering lots of individual "Error connecting to the API" messages in each 'row' of the summary list.
I actually think this would be more useful - if a little ugly - as if some of the GdsApi requests succeed and some time out, we will at least get the information for the ones that succeeded.

Shall I proceed with moving the error handling to the individual views?

Reply:
Ah yes, thanks for pointing out the subtlety here! I remember we had a similar issue when this got implemented originally. It's likely they will all succeed or not, so the original concern was about not being repetitive. I had a chat with @kevindew yesterday and we agreed the individual messages would be OK, although we could look at making them shorter.

Go for it 👍

Reply:
Done 👍